packages/firebase/index.d.ts

```diff
@@ -7209,6 +7209,43 @@ declare namespace firebase.firestore {
      */
     waitForPendingWrites(): Promise<void>;
 
+    /**
+   * Attaches a listener for a snapshots-in-sync event. The snapshots-in-sync
+   * event indicates that all listeners affected by a given change have fired,
+   * even if a single server-generated change affects multiple listeners.
+   *
+   * NOTE: The snapshots-in-sync event only indicates that listeners are in sync
+   * with each other, but does not relate to whether those snapshots are in sync
+   * with the server. Use SnapshotMetadata in the individual listeners to
+   * determine if a snapshot is from the cache or the server.
+   *
+   * @param observer A single object containing `next` and `error` callbacks.
+   * @return An unsubscribe function that can be called to cancel the snapshot
+   * listener.
+   */
+  onSnapshotsInSync(observer: {
+    next?: (value: void) => void;
+    error?: (error: Error) => void;
+    complete?: () => void;
+  }): () => void;
+
+  /**
+   * Attaches a listener for a snapshots-in-sync event. The snapshots-in-sync
+   * event indicates that all listeners affected by a given change have fired,
+   * even if a single server-generated change affects multiple listeners.
+   *
+   * NOTE: The snapshots-in-sync event only indicates that listeners are in sync
+   * with each other, but does not relate to whether those snapshots are in sync
+   * with the server. Use SnapshotMetadata in the individual listeners to
+   * determine if a snapshot is from the cache or the server.
+   *
+   * @param onSync A callback to be called every time all snapshot listeners are
+   * in sync with each other.
+   * @return An unsubscribe function that can be called to cancel the snapshot
+   * listener.
+   */
+  onSnapshotsInSync(onSync: () => void): () => void;
```

Comment:
So for onSnapshot() we seem to have 4 overloads but a single doc comment describing the method.  Can we do that here too?

Reply:
Where is that? I saw that each `onSnapshot()` [link](https://github.com/firebase/firebase-js-sdk/blob/master/packages/firebase/index.d.ts#L7758) had it own doc comment so that's what I did as well.

Reply:
OHHH...  Interesting.  I was looking at firestore-types/index.d.ts instead of firebase/index.d.ts.  see here: https://github.com/firebase/firebase-js-sdk/blob/66655d0ba42c2ebf153bd950d73612f0a1533c7f/packages/firestore-types/index.d.ts#L1174

I thought our types were basically in-sync in both files, but I guess not.  Hrm.  I'm guessing @hsubox76 had to separate them when transitioning our public docs to be generated from the .d.ts files...  in which case we should probably update firestore-types/index.d.ts to match.  If the files are out-of-sync now, they'll just get worse over time.